BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,364 @@
+            }
+        }
+          .keyboardType(.numberPad)
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkChainNameTitle)
+          .textCase(.none)
+      ) {
+        NetworkTextField(placeholder: Strings.Wallet.customNetworkChainNamePlaceholder,
+                         item: $model.networkName) { newValue in
+          if newValue.count < 1 {
+            model.networkName.error = Strings.Wallet.customNetworkEmptyErrMsg
+          } else {
+            model.networkName.error = nil
+          }
+        }
+          .disableAutocorrection(true)
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkSymbolNameTitle)
+          .textCase(.none)
+      ) {
+        NetworkTextField(placeholder: Strings.Wallet.customNetworkSymbolNamePlaceholder,
+                         item: $model.networkSymbolName) { newValue in
+          if newValue.count < 1 {
+            model.networkSymbolName.error = Strings.Wallet.customNetworkEmptyErrMsg
+          } else {
+            model.networkSymbolName.error = nil
+          }
+        }
+          .disableAutocorrection(true)
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkSymbolTitle)
+          .textCase(.none)
+      ) {
+        NetworkTextField(placeholder: Strings.Wallet.customNetworkSymbolPlaceholder,
+                         item: $model.networkSymbol) { newValue in
+          if newValue.count < 1 {
+            model.networkSymbol.error = Strings.Wallet.customNetworkEmptyErrMsg
+          } else {
+            model.networkSymbol.error = nil
+          }
+        }
+          .disableAutocorrection(true)
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkCurrencyDecimalTitle)
+          .textCase(.none)
+      ) {
+        NetworkTextField(placeholder: Strings.Wallet.customNetworkCurrencyDecimalPlaceholder,
+                         item: $model.networkDecimals) { newValue in
+          if newValue.isEmpty {
+            model.networkDecimals.error = Strings.Wallet.customNetworkEmptyErrMsg
+          } else if let intValue = Int(newValue), intValue > 0 {
+            model.networkDecimals.error = nil
+          } else {
+            model.networkDecimals.error = Strings.Wallet.customNetworkCurrencyDecimalErrMsg
+          }
+        }
+          .keyboardType(.numberPad)
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkRpcUrlsTitle)
+          .textCase(.none)
+      ) {
+        ForEach(model.rpcUrls.indices, id: \.self) { index in
```

Comment:
Identifying based on index is a bit dangerous, and will result in changes to the underlying URL in the model not actually updating the views. If we can we should replace this with non-index based IDs